### PR TITLE
add new nextjs typings with samples

### PR DESCRIPTION
**Pull request by calendee**
The current `examples/with-typescript` is not using the latest type definitions currently available on DefinitelyTyped project for next.

Added new list page examples that demonstrate how to use the new Types for both stateless functional components and classes.  Also modified examples for list to demonstrate typings for `getInitialProps`.

### CODE REVIEW

**Review by timneutkens — APPROVED** [review]

File: examples/with-typescript/components/ListItem.tsx [hunk 1]
```diff
@@ -1,12 +1,12 @@
 import * as React from 'react'
-import {DataObject} from "./List";
+import IDataObject from '../interfaces'
 
-export interface Props {
-    data: DataObject
+export interface IProps {
```
Comment (resir014) [inline examples/with-typescript/components/ListItem.tsx]: Can we change these `interface`s to `type`s instead?

File: examples/with-typescript/components/Layout.tsx [hunk 2]
```diff
@@ -3,29 +3,30 @@ import Link from 'next/link'
...
-        <Link href='/about'><a>About</a></Link> | {' '}
-      </nav>
-    </header>
-    {children}
-    <footer>
-        <hr/>
-        <span>I'm here to stay (Footer)</span>
-    </footer>
-  </div>
+    <div>
```
Comment (resir014) [inline examples/with-typescript/components/Layout.tsx]: Let's keep the two-space indentation in all files to keep the diffs clean.

File: examples/with-typescript/pages/list-class.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import React from 'react'
+import { NextContext } from 'next'
+
+import Layout from '../components/Layout'
+import List from '../components/List'
+import IDataObject from '../interfaces'
+
+interface IListClassProps {
```
Comment (resir014) [inline examples/with-typescript/pages/list-class.tsx]: Same here for `type`s v. `interface`s.

**Review by resir014 — CHANGES_REQUESTED** [review]
Hey, this looks good to me. Just a few styling changes needed, though.

**Comment by calendee** [comment]
Anyone know how/why coverage would have changed with this PR?  I did delete an unused file `examples/with-typescript/pages/list.tsx`.  However, I don't see any coverage tests for it.  

When I view the coverage changes, I don't see any changes that affect the `examples` directory at all.

Any idea on how to fix this or overcome the required checks for it?

**Review by resir014 — APPROVED** [review]
Yep, looks good!

**Review by timneutkens — APPROVED** [review]

**Pull request merged**